tools/tests/test_nd_freeze

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+# Temporary scratch directory to save generated files.
+TMP_DIR="tmp_$((1 + RANDOM % 10000))"
+
+# Run common commands for each test at start of tests
+#
+# Parameters
+# ----------
+# distro : string
+#   Valid values are "debain" or "ubuntu"
+# release : string
+#   Code name for distro release. (e.g. jessie)
+# date : string
+#   Date at which the snapshot of the OS is wanted
+# keep_sources : string (Optional)
+#   Pass the string "KEEP_SOURCES" to run nd_freeze with the --keep-original-apt-sources switch
+#
+function test_setup {
+    distro=$1
+    release=$2
+    date=$3
+    keep_sources=${4:-no}
+
+    if [ "$keep_sources" = "no" ]; then
+        keep_sources_param=""
+    else
+        keep_sources_param="--keep-original-apt-sources"
+    fi
+
+    mkdir -p $TMP_DIR
+
+    docker run -it --rm -v $PWD/..:/home ${distro}:${release} /bin/bash -c "
+        apt-get update
+        apt-get install -y wget
+        wget -O- http://neuro.debian.net/lists/${release}.us-nh.full | tee /etc/apt/sources.list.d/neurodebian.sources.list
+        apt-key adv --recv-keys --keyserver hkp://pool.sks-keyservers.net:80 0xA5D32F012649A5A9
+        apt-get update
+        /home/nd_freeze $keep_sources_param $date
+        if [ -f /etc/apt/sources.list ]; then
+            cp /etc/apt/sources.list /home/tests/$TMP_DIR/sources.list
+        fi
+        if [ -f /etc/apt/sources.list.original ]; then
+            cp /etc/apt/sources.list.original /home/tests/$TMP_DIR/sources.list.original
+        fi
+        if [ -f /etc/apt/sources.list.d/neurodebian.sources.list ]; then
+            cp /etc/apt/sources.list.d/neurodebian.sources.list /home/tests/$TMP_DIR/neurodebian.sources.list
+        fi
+        if [ -f /etc/apt/sources.list.d/neurodebian.sources.list.original ]; then
+            cp /etc/apt/sources.list.d/neurodebian.sources.list.original /home/tests/$TMP_DIR/neurodebian.sources.list.original
+        fi
+        if [ -f /etc/apt/sources.list.d/snapshots.sources.list ]; then
+            cp /etc/apt/sources.list.d/snapshots.sources.list /home/tests/$TMP_DIR/snapshots.sources.list
+        fi
+    " > $TMP_DIR/stdout
+}
+
+# Run commands that are run at the end of each test run.
+function test_teardown {
+    rm -rf $TMP_DIR
+}
+
+# Test to see if a line is in a given file.
+#
+# Parameters
+# ----------
+# file : string
+#   File to search for the given string
+#
+# line : string
+#   Line the we expect to be in the file.
+#
+# match_mode : string
+#   Enter "partial_match" if the line given is not a complete line but a line fragment
+#
+function assert_line_in_file {
+    file=$1
+    line=$2
+    match_mode=${3:-full}
+
+    if [ "$match_mode" = "full" ]; then
+        match_param="x"
+    else
+        match_param=""
+    fi
+
+    if grep -Fq$match_param "$line" "$TMP_DIR/$file"; then
+        echo "SUCCESS: line found"
```

Comment:
May be add which line in what file?  otherwise it is boring list of "SUCCESSes" without a detail to differ them